src/transformers/models/ibert/modeling_ibert.py

```diff
@@ -0,0 +1,1418 @@
+        return hidden_states, hidden_states_scaling_factor
+
+
+# Referred from transformers.models.bert.modeling_bert.BertAttention
+class IBertAttention(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.quant_mode = config.quant_mode
+        self.self = IBertSelfAttention(config)
+        self.output = IBertSelfOutput(config)
+        self.pruned_heads = set()
+
+    def prune_heads(self, heads):
+        if len(heads) == 0:
+            return
+        heads, index = find_pruneable_heads_and_indices(
+            heads, self.self.num_attention_heads, self.self.attention_head_size, self.pruned_heads
+        )
+
+        # Prune linear layers
+        self.self.query = prune_linear_layer(self.self.query, index)
+        self.self.key = prune_linear_layer(self.self.key, index)
+        self.self.value = prune_linear_layer(self.self.value, index)
+        self.output.dense = prune_linear_layer(self.output.dense, index, dim=1)
+
+        # Update hyper params and store pruned heads
+        self.self.num_attention_heads = self.self.num_attention_heads - len(heads)
+        self.self.all_head_size = self.self.attention_head_size * self.self.num_attention_heads
+        self.pruned_heads = self.pruned_heads.union(heads)
+
+    def forward(
+        self,
+        hidden_states,
+        hidden_states_scaling_factor,
+        attention_mask=None,
+        head_mask=None,
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
+        past_key_value=None,
+        output_attentions=False,
+    ):
+        self_outputs, self_outputs_scaling_factor = \
+            self.self(
+                hidden_states,
+                hidden_states_scaling_factor,
+                attention_mask,
+                head_mask,
+                encoder_hidden_states,
+                encoder_attention_mask,
+                past_key_value,
+                output_attentions,
+            )
+        attention_output, attention_output_scaling_factor = self.output(
+                self_outputs[0], self_outputs_scaling_factor[0],
+                hidden_states, hidden_states_scaling_factor)
+        outputs = (attention_output,) + self_outputs[1:]  # add attentions if we output them
+        outputs_scaling_factor = (attention_output_scaling_factor,) + \
+                self_outputs_scaling_factor[1:]
+        return outputs, outputs_scaling_factor
+
+
+# Referred from transformers.models.bert.modeling_bert.BertIntermediate
+class IBertIntermediate(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.quant_mode = config.quant_mode
+        self.act_bit = 8
+        self.weight_bit = 8
+        self.bias_bit = 32
+        self.dense = QuantLinear(self.weight_bit, bias_bit=self.bias_bit, 
+                quant_mode=self.quant_mode, per_channel=True)
+        self.dense.set_param(nn.Linear(config.hidden_size, config.intermediate_size))
+
+        assert config.hidden_act == 'gelu'
```

Comment:
Could we add messages to the assertions so that we know what's the issue if it crashes?